pkg/specgen/container_validate.go

```diff
@@ -135,38 +133,6 @@ func (s *SpecGenerator) Validate() error {
 	// default:
 	//	return errors.New("unrecognized option for cgroups; supported are 'default', 'disabled', 'no-conmon'")
 	// }
-	invalidUlimitFormatError := errors.New("invalid default ulimit definition must be form of type=soft:hard")
-	// set ulimits if not rootless
-	if len(s.ContainerResourceConfig.Rlimits) < 1 && !rootless.IsRootless() {
```

Comment:
Why did you remove this? This sets the limit based on containers.conf defaults so we need to keep it. Although doing this in Validate() looks wrong.

Reply:
I'm under the impression that it's already set by
https://github.com/containers/podman/blob/77bd041a9665373b23a1cab77d4bffe28514649a/pkg/specgenutil/specgen.go#L771

Doing a quick test it looks like a ulimit from containers.conf still makes it into a container, so this was redundant. However it's in there twice right now. So I think the containers.conf defaults also make it to:
https://github.com/containers/podman/blob/77bd041a9665373b23a1cab77d4bffe28514649a/pkg/specgen/generate/container_create.go#L34

Reply:
The issue with the limits being in the list twice is fixed.